### PR TITLE
Prevent NPE in gitlab client if connection cannot be estabilished

### PR DESCRIPTION
**Pull request by mmorhun**
### Checklist:
 - PR has reference to the issue(s) it resolves
 - Write / update unit tests
 - Write / update integration (envtest) tests
 - Ensure there is an issue for e2e tests if needed
 - Ensure `make test` passes
 - Ensure test coverage hasn't decreased
 - Test code changes manually
 - Update readme and documentation
 - Write PR description that highlights overall changes and add usage examples if applicable

### CODE REVIEW

File: controllers/component_build_controller_unit_test.go [hunk 1]
```diff
@@ -1641,6 +1641,11 @@ func TestGetGitProvider(t *testing.T) {
 			componentRepoUrl: "12345",
 			expectError:      true,
 		},
+		{
+			name:             "should return error if git source URL is empty",
+			componentRepoUrl: "",
+			expectError:      true,
+		},
```
Comment (chmeliik) [inline controllers/component_build_controller_unit_test.go:1648]: For my understanding, this test case is not related to the null pointer exception right?

**Review by chmeliik — APPROVED** [review]
LGTM :+1:

File: controllers/component_build_controller_unit_test.go (hunk 1 above)
Comment (mmorhun) [reply controllers/component_build_controller_unit_test.go:1648]: Yes, right
